packages/jsapi-shim/src/dh.types.ts

```diff
@@ -0,0 +1,682 @@
+  readonly isMinorTicksVisible: boolean;
+  readonly isMajorTicksVisible: boolean;
+  readonly minorTickCount: number;
+  readonly gapBetweenMajorTicks: number;
+  readonly majorTickLocations: number[];
+  readonly tickLabelAngle: number;
+  readonly isInvert: boolean;
+  readonly isTimeAxis: boolean;
+
+  /**
+   * Indicate the density and range of data that the UI needs for this axis, across any series which
+   * draws on this axis. Ignored for non-time series data, for non-line series.
+   * @param pixelCount the approx number of pixels wide
+   * @param min the optional minimum value visible on this axis - even if specified, smaller values may
+   * be returned, to ensure that lines drawn off the screen.
+   * @param max the optional max value visible on this axis. If min is specified, max is also expected.
+   */
+  range(pixelCount?: number, min?: any, max?: any): void;
+}
+
+export interface SeriesDataSource {
+  readonly axis: Axis;
+  readonly type: SourceType;
+  readonly columnName: string;
+}
+
+export interface OneClick {
+  readonly columns: { name: string; type: string }[];
+  readonly isRequireAllFiltersToDisplay: boolean;
+
+  setValueForColumn(columnName: string, value: any): void;
+  getValueForColumn(columnName: string): any;
+}
+
+export interface Column {
+  readonly type: string;
+  readonly name: string;
+  readonly description: string;
+  readonly constituentType: string;
+
+  readonly isPartitionColumn: boolean;
+
+  filter(): FilterValue;
+  sort(): Sort;
+}
+export interface FilterValue {
+  ofString(input: object): FilterValue;
+  ofNumber(input: number): FilterValue;
+  ofNumber(input: LongWrapper): FilterValue;
+  ofNumber(input: object): FilterValue;
+  ofBoolean(input: boolean): FilterValue;
+
+  eq(value: FilterValue): FilterCondition;
+  eqIgnoreCase(value: FilterValue): FilterCondition;
+  notEq(value: FilterValue): FilterCondition;
+  notEqIgnoreCase(value: FilterValue): FilterCondition;
+  greaterThan(value: FilterValue): FilterCondition;
+  lessThan(value: FilterValue): FilterCondition;
+  greaterThanOrEqualTo(value: FilterValue): FilterCondition;
+  lessThanOrEqualTo(value: FilterValue): FilterCondition;
+  inIgnoreCase(value: FilterValue): FilterCondition;
+  notIn(value: FilterValue): FilterCondition;
+  notInIgnoreCase(value: FilterValue): FilterCondition;
+  contains(value: FilterValue): FilterCondition;
+  isFalse(): FilterCondition;
+  isTrue(): FilterCondition;
+  isNull(): FilterCondition;
+  invoke(method: string, ...args: FilterValue[]): FilterCondition;
+}
+export interface FilterCondition {
+  invoke(method: string, ...args: FilterValue[]): FilterCondition;
+  search(value: FilterValue, columns?: FilterValue[]): FilterCondition;
+
+  not(): FilterCondition;
+  and(first: FilterCondition, ...rest: FilterCondition[]): FilterCondition;
+  or(first: FilterCondition, ...rest: FilterCondition[]): FilterCondition;
+
+  toString(): string;
+  // columns: Column[] //doesnt work
```

Comment:
Just take this out, I don't know why it's even in the code, it always returns an empty array.
```suggestion
```